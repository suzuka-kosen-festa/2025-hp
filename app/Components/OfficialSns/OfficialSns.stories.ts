import type { Meta, StoryObj } from "@storybook/react";
import OfficialSns from "./OfficialSns";

const meta = {
  title: "Components/OfficialSns",
  component: OfficialSns,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
} satisfies Meta<typeof OfficialSns>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {};

export const CustomLinks: Story = {
  args: {
    xUrl: "https://x.com/",
    instagramUrl: "https://instagram.com/",
    facebookUrl: "https://facebook.com/",
  },
};
