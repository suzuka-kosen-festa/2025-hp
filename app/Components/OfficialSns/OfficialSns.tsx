import type { ReactNode } from "react";
import FacebookIcon from "@mui/icons-material/Facebook";
import InstagramIcon from "@mui/icons-material/Instagram";
import XIcon from "@mui/icons-material/X";
import { IconButton, Stack } from "@mui/material";

interface OfficialSnsProps {
  xUrl?: string
  instagramUrl?: string
  facebookUrl?: string
}

function OfficialSns({
  xUrl = "https://x.com/KOSENFESTA",
  instagramUrl = "https://www.instagram.com/kosenfesta/",
  facebookUrl = "https://www.facebook.com/profile.php?id=61556514660247",
}: OfficialSnsProps): ReactNode {
  return (
    <Stack direction="row" spacing={1}>
      <IconButton
        aria-label="X"
        href={xUrl}
        target="_blank"
        rel="noopener noreferrer"
        sx={{ color: "white" }}
      >
        <XIcon />
      </IconButton>
      <IconButton
        aria-label="Instagram"
        href={instagramUrl}
        target="_blank"
        rel="noopener noreferrer"
        sx={{ color: "white" }}
      >
        <InstagramIcon />
      </IconButton>
      <IconButton
        aria-label="Facebook"
        href={facebookUrl}
        target="_blank"
        rel="noopener noreferrer"
        sx={{ color: "white" }}
      >
        <FacebookIcon />
      </IconButton>
    </Stack>
  );
}

export default OfficialSns;
