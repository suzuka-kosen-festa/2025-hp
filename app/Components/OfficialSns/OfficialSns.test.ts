// @vitest-environment jsdom

import { composeStories } from "@storybook/react";
import { expect, it } from "vitest";
import * as stories from "./OfficialSns.stories";

const { Default } = composeStories(stories);

it("officialSns snapshot", async () => {
  await Default.run();
  expect(document.body.firstChild).toMatchSnapshot();
});
