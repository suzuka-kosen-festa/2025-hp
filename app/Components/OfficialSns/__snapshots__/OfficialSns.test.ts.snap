// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`officialSns snapshot 1`] = `
<div>
  <div
    class="MuiStack-root css-niqf4j-MuiStack-root"
  >
    <a
      aria-label="X"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-j7dfm5-MuiButtonBase-root-MuiIconButton-root"
      href="https://x.com/KOSENFESTA"
      rel="noopener noreferrer"
      tabindex="0"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="XIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18.244 2.25h3.308l-7.227 8.26 8.502 11.24H16.17l-5.214-6.817L4.99 21.75H1.68l7.73-8.835L1.254 2.25H8.08l4.713 6.231zm-1.161 17.52h1.833L7.084 4.126H5.117z"
        />
      </svg>
    </a>
    <a
      aria-label="Instagram"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-j7dfm5-MuiButtonBase-root-MuiIconButton-root"
      href="https://www.instagram.com/kosenfesta/"
      rel="noopener noreferrer"
      tabindex="0"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="InstagramIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
        />
      </svg>
    </a>
    <a
      aria-label="Facebook"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-j7dfm5-MuiButtonBase-root-MuiIconButton-root"
      href="https://www.facebook.com/profile.php?id=61556514660247"
      rel="noopener noreferrer"
      tabindex="0"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="FacebookIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
        />
      </svg>
    </a>
  </div>
</div>
`;
